samples/AspNetCoreSseServer/Attributes/LimitCalls.cs

```diff
@@ -0,0 +1,31 @@
+﻿using ModelContextProtocol.Core;
+using ModelContextProtocol.Protocol;
+using ModelContextProtocol.Server;
+
+namespace AspNetCoreSseServer.Attributes;
+
+public class LimitCalls(int maxCalls, int order = 0) : ToolFilter(order)
+{
+    private int _callCount;
+
+    public override ValueTask<CallToolResult>? OnToolCalling(Tool tool, RequestContext<CallToolRequestParams> context)
+    {
+        _callCount++;
```

Comment:
This technically has a race condition, and should be using Interlocked.Add

Reply:
Thanks. I'll update the example.